### PR TITLE
Add support for filtering on ID's for collections.

### PR DESCRIPTION
**Pull request by nickygerritsen**

### CODE REVIEW

File: Contest_API.md [hunk 1]
```diff
@@ -371,6 +371,15 @@ attributes that are:
...
     `null` value or because it was not implemented.
 
+### Filtering
+
+Endpoints that return a JSON array must allow filtering on any
+attribute with type ID (except the `id` field) by passing it as a
+query argument. For example, clarifications can be filtered on the
+recipient by passing `to_team_id=X`. To filter on a `null` value,
+pass an empty string, i.e. `to_team_id=`. It must be possible to filter
+on mulitple fields simultaneously.
```
Comment (eldering) [inline Contest_API.md:381]: Typo: "multiple".
Maybe a bit more explicit: "It must be possible to filter on multiple different fields simultaneously, with the meaning that all conditions must be met (they are logically `AND`ed)."

File: Contest_API.md [hunk 2]
```diff
@@ -371,6 +371,15 @@ attributes that are:
     In the latter case that can either be because it was a left out
     `null` value or because it was not implemented.
 
+### Filtering
+
+Endpoints that return a JSON array must allow filtering on any
+attribute with type ID (except the `id` field) by passing it as a
```
Comment (eldering) [inline Contest_API.md:377]: What about attributes that are an array of IDs, like `group_ids` in the teams endpoint and `team_ids` in awards?
I'd argue to start simple and not support these yet.

Comment (nickygerritsen) [reply Contest_API.md:377]: Thats what I only mention ID. I could explicitly state that filtering on array of ID does not have to be supported

Comment (nickygerritsen) [reply Contest_API.md:377]: Added a note about array of ID

File: Contest_API.md [hunk 3]
```diff
@@ -371,6 +371,18 @@ attributes that are:
...
+query argument. For example, clarifications can be filtered on the
+recipient by passing `to_team_id=X`. To filter on a `null` value,
+pass an empty string, i.e. `to_team_id=`. It must be possible to
+filter on multiple different fields simultaneously, with the
+meaning that all conditions must be met (they are logically `AND`ed).
+Note that filtering on any other field, including fields with the type
+array of ID, does not have to be supported.
```
Comment (niemela) [inline Contest_API.md:384]: "array of ID" should be backticked.

**Review by niemela — CHANGES_REQUESTED** [review]

File: Contest_API.md (hunk 3 above)
Comment (nickygerritsen) [reply Contest_API.md:384]: Then `ID` above should be as well? It is also not backticked in the tables though

Comment (niemela) [reply Contest_API.md:384]: > Then `ID` above should be as well? 

Good point.

> It is also not backticked in the tables though

Yeah... You know what, let's not...

(And some point we should think through what exactly should be quoted how and then make sure we are consistent. But that's just editing, and can trivially be done later. Let's not block relevant updates on that.)

**Review by niemela — APPROVED** [review]

**Review by eldering — APPROVED** [review]

**Pull request merged**